modules/vision/src/key-point/vpKeyPoint.cpp

```diff
@@ -1920,9 +1920,11 @@ void vpKeyPoint::initDetector(const std::string &detectorName)
 
   bool detectorInitialized = false;
   if (!usePyramid) {
-    detectorInitialized = (m_detectors[detectorNameTmp] != NULL);
+    //if not null and to avoid warning C4800: forcing value to bool 'true' or 'false' (performance warning)
+    detectorInitialized = !!m_detectors[detectorNameTmp];
   } else {
```

Comment:
Could be also `!m_detectors[detectorNameTmp].empty()` that I found more explicit